tests/other_modes.rs

```diff
@@ -229,10 +229,9 @@ fn edge_oneshot_triggered() {
         .wait(&mut events, Some(Duration::from_secs(10)))
         .unwrap();
 
-    assert_eq!(
-        events.iter().collect::<Vec<_>>(),
-        [Event::readable(reader_token)]
-    );
+    assert_eq!(events.len(), 1);
+    assert!(events.iter().next().unwrap().readable);
+    assert!(!events.iter().next().unwrap().writable);
```

Comment:
this pattern occurs multiple times in this commit but it looks like a code smell. imo it would be better find a way to do both of these asserts on the same flag variable. (e.g. by retrieving `events.iter().next().unwrap()` before these asserts and reuse the result)